index.html

```diff
@@ -4683,82 +4683,134 @@ <h2>Internationalization Considerations</h2>
 software agents rendering the property can deterministically identify the
-language in use.
+various languages in use. JSON implementations following the
+<code>@context</code> ordering requirements in this specification will also be
+able to benefit from this feature.
           </p>
         </li>
       </ul>
     </section>
 
     <section class="informative">
-      <h3>Text direction</h3>
-
-      <p>
-When expressing human-readable text content in the data model, it is important
-to be able to explicitly encode the text direction. JSON and its derived formats
-use the UTF-8 encoding of [[Unicode]] for serializing and transmitting text.
-This enables text direction to be directly determined in simple cases,
-but the mechanism is insufficient for general use.
-      </p>
+      <h3>Varied Text Directions</h3>
 
       <p>
-[[!JSON-LD]] supports the expression of text layout information using the
-<code>rdf:HTML</code> string literal type, as described in [[!JSON-LD]] 1.0,
-<a href="https://www.w3.org/TR/json-ld11/#typed-values">Section 4.2.1: Typed Values</a>.
+When multiple text directions are utilized in the same value, a more
+complex markup mechanism is required. [[!JSON-LD]] supports the expression of
+text layout information using the <code>rdf:HTML</code> string literal type,
+as described in [[!JSON-LD]], <a href="https://www.w3.org/TR/json-ld11/#typed-values">Section 4.2.1: Typed Values</a>.
 This mechanism enables developers to use the internationalization features
-present in HTML to express language information. For example, consider the
-following key-value pair:
-
-      <pre>
-"nameHtml": "&lt;span dir="rtl" lang="ar">HTML و CSS: تصميم و إنشاء مواقع الويب&lt;/span>"
+present in HTML to express text direction information. For example, consider the
+following key-value pair (line breaks added for readability):
+
+      <pre class="example nohighlight" title="Expressing multiple text directions in single text string">
+"descriptionHtml": "&lt;span lang="en"  <span class="highlight">dir="ltr"</span>>The country names include
+                      &lt;span lang="ar" <span class="highlight">dir="rtl"</span>>مصر&lt;/span> and
+                      &lt;span lang="he"  <span class="highlight">dir="rtl"</span>>ישראל&lt;/span>.
+                    &lt;/span>"
       </pre>
 
       <p>
 If the JSON-LD Context states that any value associated with the
 <code>nameHtml</code> property is of type <code>rdf:HTML</code>, then software
 agents have sufficient information to deterministically identify the text
-direction of the language.
+direction of the language, even if the direction changes within the same string.
+JSON implementations following the <code>@context</code> ordering requirements
+in this specification will also be able to benefit from this feature.
       </p>
 
     </section>
 
     <section class="informative">
-      <h3>Ruby Text</h3>
+      <h3>Pronunciation</h3>
       <p>
 In some situations it is common to use
```

Comment:
I think I commented on this before: this seems like a stretch that you'd be better off ignoring (leaving out altogether). In the same way that you wouldn't put text here saying "In some situations it is common to use markup to indicate emphasis and you might need the `<b>` and `<strong>` tags". I love me some Ruby markup and implementers who want to exchange data with embedded ruby markup are totally welcome to do so. But it is necessary to call it out specially?

Reply:
I merged the Ruby bits into the "Complex Markup" section to give people an idea on when to use `rdf:HTML`, and when to use the other mechanism. I still don't mention BiDi specifically, but it feels like something that happens automatically when a developer copy-pastes text or types it in... and thus, we don't need to mention it?

Reply:
After consulting w/ @dlongley, he noted several attacks that can be pulled off by the use of HTML + Verifiable Credentials that are scary enough to rewrite that section noting the attacks. We're discouraging the use of HTML to express language and base direction unless absolutely necessary. Please re-read the section given the new, new, new complete rewrite.